### DISCUSSION TITLE
Issues running code-server locally.

### DISCUSSION THREAD

**Discussion by emailvenkatm** [coder/code-server#3327] [Q&A] [answered]
Hello,

I am able to start the code-server locally using install and it is working perfectly. But I try to run it by closing the repo and running yarn & yarn watch. It is loading fine but not able to open integrated terminal.  Getting the below error. 

Tried troubleshooting this by switching to different node versions but no luck. Can some one please help me on how to fix this issue.

Error:

The terminal process failed to launch: A native exception occurred during launch (The module '~/code-server/lib/vscode/node_modules/node-pty/build/Release/pty.node' was compiled against a different Node.js version using NODE_MODULE_VERSION 87. This version of Node.js requires NODE_MODULE_VERSION 83. Please try re-compiling or re-installing the module (for instance, using `npm rebuild` or `npm install`).).

**Comment by code-asher** [comment, marked as answer]
I usually fix this by going into `lib/vscode` and running `npm rebuild`.

It seems to happen when the modules were installed with one version of Node and code-server is ran with a different version but the weird thing is that it happens to me even if I only have one version of Node installed so I'm not sure what's going on.

**Reply by jsjoeio** [reply]
Same happens to me ^. We need to document this in `CONTRIBUTING.md`. I'll make a note.

**Reply by jsjoeio** [reply]
https://github.com/cdr/code-server/issues/3336

**Comment by swapnak152** [comment]
Thank you so much! It worked live a charm

**Comment by wangqingshun** [comment]
I want to run the project locally in windows, but I get an error. How do you run it locally after you pull the code?

**Reply by code-asher** [reply]
You can either use `yarn && yarn watch` or build it (see https://github.com/cdr/code-server/blob/main/docs/CONTRIBUTING.md#development-workflow).

However I'm not sure our scripts will run on Windows. I don't think anyone has tried yet.
